Client/DispatchQueueProtocol.swift

```diff
@@ -0,0 +1,32 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0
+
+import Foundation
+
+enum DispatchQueueContext: Equatable {
+    case main
+    case global(DispatchQoS.QoSClass)
+}
+
+protocol DispatchQueueProtocol {
+    func shared(_ context: DispatchQueueContext) -> DispatchQueue
+    func async(context: DispatchQueueContext, _ work: @escaping @convention(block) () -> Void)
+}
+
+extension DispatchQueueProtocol {
+    func shared(_ context: DispatchQueueContext) -> DispatchQueue {
+        switch context {
+        case .main:
+            return DispatchQueue.main
+        case .global(let qos):
+            return DispatchQueue.global(qos: qos)
+        }
+    }
+}
+
+class MozillaDispatchQueue: DispatchQueueProtocol {
+    func async(context: DispatchQueueContext, _ work: @escaping @convention(block) () -> Void) {
+        shared(context).async(execute: work)
+    }
+}
```

Comment:
From my current understanding, this change was brought up so the queue can be passed in as a dependency during tests, and then assert which thread we're on? 

If so, I would refrain for now from adding a new queue protocol at the moment, reason being that we might go with different async solutions in the near future. The more solutions we have, the more complicated it gets to move away from DispatchQueue. For the time being, I would just use an expectation on `test_updateData_tabTrayGroupsDisabled_stubRecentTabsWithStartingURLs_max2` test to make the needed assertions.

Reply:
Okey dokey. I can walk back these changes and change these tests to use expectations.

Reply:
Cool, thanks and sorry for trouble!

Reply:
It's no worries. I totally understand. I definitely wouldn't wanna muddy the waters if y'all are fixing to change how you capture async behaviors. Maybe once y'all do decide, I can go back through this class and clean things up again.